### ISSUE TITLE
Option to preserve timestamps on downloaded files

### ISSUE THREAD

**Issue by hickford** [SaltwaterC/http-request#6]
Hi. It would be useful to have an option to preserve the timestamps when downloaded files are saved to this. This happens to be default behaviour of `wget` and other download software http://www.gnu.org/software/wget/manual/wget.html

> By default, when a file is downloaded, it's timestamps are set to match those from the remote file. This allows the use of ‘--timestamping’ on subsequent invocations of wget. However, it is sometimes useful to base the local file's timestamp on when it was actually downloaded; for that purpose, the ‘--no-use-server-timestamps’ option has been provided.

Modern web browsers don't preserve timestamps:
- Firefox https://bugzilla.mozilla.org/show_bug.cgi?id=178506
- Chrome http://code.google.com/p/chromium/issues/detail?id=4574

**Comment by SaltwaterC** [comment]
I'll take a stab at figuring out which of the fs-related call is required in order to implement this. Any suggestion is welcome.

**Comment by hickford** [comment]
Hi Stefan. Thanks for your interest. Here's how I did it in my script:

```
httpget = require('http-get')
fs = require('fs')   
await httpget.get({url:url},destination,defer(err,result))
timestamp = new Date(result.headers['last-modified'])
await fs.utimes(destination,timestamp,timestamp,defer())
```

Thinking about it, I concluded that preserving timestamps has a niche use case (viz. mirroring) and while a vital feature to `wget` isn't necessary in a library since one can take the timestamp from the response header as above. I am closing the issue. Of course, do as you like.
